webv2/web.go

```diff
@@ -377,36 +381,56 @@ func getSourceProfileConfig(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(sourceProfileConfig)
 }
 
+func setDatastreamDetailsForShardedMigrations(w http.ResponseWriter, r *http.Request) {
+	sessionState := session.GetSessionState()
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var datastreamConfigUnmarshaler DatastreamConfigUnmarshaler
```

Comment:
why do we need DatastreamConfigUnmarshaler , can we directly not use profiles.DatastreamConfig?

Reply:
It was throwing some error, I am not sure why that does not work, it would require deeper investigation. We also do the same thing for dataflow so just followed that here for now.